pkg/hubble/filters/httpheader_test.go

```diff
@@ -0,0 +1,303 @@
+
+		{
+			name: "header_key_match",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						HttpHeader: []*flowpb.HTTPHeader{
+							{Key: "Content_Length",
+								Value: "",
+							}},
+					},
+				},
+				ev: []*v1.Event{
+					httpFlow(&flowpb.HTTP{
+						Headers: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					}),
+				},
+			},
+
+			want: true,
+		},
+
+		{
+			name: "header_value_match",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						HttpHeader: []*flowpb.HTTPHeader{
+							{Key: "",
+								Value: "162",
+							}},
+					},
+				},
+				ev: []*v1.Event{
+					httpFlow(&flowpb.HTTP{
+						Headers: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					}),
+				},
+			},
+
+			want: true,
+		},
+		{
+			name: "header_key_and_value_match",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						HttpHeader: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					},
+				},
+				ev: []*v1.Event{
+					httpFlow(&flowpb.HTTP{
+						Headers: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					}),
+				},
+			},
+
+			want: true,
+		},
+
+		{
+			name: "header_key_mismatch",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						HttpHeader: []*flowpb.HTTPHeader{
+							{Key: "Connection",
+								Value: "162",
+							},
+						},
+					},
+				},
+				ev: []*v1.Event{
+					httpFlow(&flowpb.HTTP{
+						Headers: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					}),
+				},
+			},
+
+			want: false,
+		},
+
+		{
+			name: "header_value_mismatch",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						HttpHeader: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "1652",
+							},
+						},
+					},
+				},
+				ev: []*v1.Event{
+					httpFlow(&flowpb.HTTP{
+						Headers: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					}),
+				},
+			},
+
+			want: false,
+		},
+
+		{
+			name: "header_key_and_value_mismatch",
+			args: args{
+				f: []*flowpb.FlowFilter{
+					{
+						HttpHeader: []*flowpb.HTTPHeader{
+							{
+								Key:   "Context_Length",
+								Value: "1652",
+							},
+						},
+					},
+				},
+				ev: []*v1.Event{
+					httpFlow(&flowpb.HTTP{
+						Headers: []*flowpb.HTTPHeader{
+							{
+								Key:   "Content_Length",
+								Value: "162",
+							},
+						},
+					}),
+				},
+			},
+
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fl, err := BuildFilterList(context.Background(), tt.args.f, []OnBuildFilter{&HTTPHeaderFilter{}})
+			assert.NoError(t, err)
+			assert.Equal(t, tt.want, fl.MatchOne(tt.args.ev))
```

Comment:
The compilation error is here: [`fl.MatchOne` expect a `*v1.Event` parameter](https://github.com/cilium/cilium/blob/76837ead2978467e1033ee40c35d7a26beadf39d/pkg/hubble/filters/filters.go#L38) but `tt.args.ev` is a slice.